test/controllers/imports_controller_test.rb

```diff
@@ -1,47 +1,67 @@
 require "test_helper"
 
 class ImportsControllerTest < ActionDispatch::IntegrationTest
-  setup_current_user { build(:user) }
-  setup { @podcast = create(:podcast) }
+  setup_current_user { build(:user, account_id: 123) }
+  setup { @podcast = create(:podcast, prx_account_uri: "/api/v1/accounts/123") }
+
+  before do
+    stub_requests
+  end
+
+  let(:feed) { Feedjira.parse(test_file("/fixtures/transistor_two.xml")) }
+  let(:import_1) { PodcastImport.create(podcast: @podcast, url: feed.url) }
+  let(:params) { {url: feed.url} }
 
   test "should get index" do
     get podcast_imports_url(@podcast)
     assert_response :success
   end
 
-  # test "should get new" do
-  #   get new_import_url
-  #   assert_response :success
-  # end
-
-  # test "should create import" do
-  #   assert_difference("Import.count") do
-  #     post imports_url, params: {import: {}}
-  #   end
+  test "authorizes index" do
+    @podcast.update(prx_account_uri: "/api/v1/accounts/456")
+    get podcast_imports_url(@podcast)
+    assert_response :forbidden
+  end
 
-  #   assert_redirected_to import_url(Import.last)
-  # end
+  test "should show import" do
+    get podcast_import_url(@podcast, import_1)
+    assert_response :success
+  end
 
-  # test "should show import" do
-  #   get import_url(@import)
-  #   assert_response :success
-  # end
+  test "authorizes show" do
+    @podcast.update(prx_account_uri: "/api/v1/accounts/456")
+    get podcast_import_url(@podcast, import_1)
+    assert_response :forbidden
+  end
 
-  # test "should get edit" do
-  #   get edit_import_url(@import)
-  #   assert_response :success
-  # end
+  test "should create import" do
+    assert_difference("PodcastImport.count") do
+      post podcast_imports_url(@podcast), params: {podcast_import: params}
+    end
 
-  # test "should update import" do
-  #   patch import_url(@import), params: {import: {}}
-  #   assert_redirected_to import_url(@import)
-  # end
+    assert_redirected_to podcast_import_url(@podcast, PodcastImport.last)
+  end
 
-  # test "should destroy import" do
-  #   assert_difference("Import.count", -1) do
-  #     delete import_url(@import)
-  #   end
+  test "authorizes creating imports" do
+    @podcast.update(prx_account_uri: "/api/v1/accounts/456")
+    post podcast_imports_url(@podcast), params: {podcast_import: params}
+    assert_response :forbidden
+  end
 
-  #   assert_redirected_to imports_url
+  # test "validates creating imports" do
```

Comment:
for some reason i can't seem to figure out how to hit the branch in the controller where an `import.save` fails and we get the expected response. but the test is written and commented out for now.

Reply:
The only validation I see on `PodcastImport` is `validates :url, presence: true`.  And you can't even get to that, because that `get_feed` method throws an error before we even try to save.

So - I think to make this test work, you need to:

- Rescue the `.get_feed` error(s) that you get from bad uris, non-rss, etc.  (May have to trial and error to see what exactly it throws for each of those bad inputs).
- In the rescue, render the flash error and return `:unprocessable_entity` (same as a model validation would have)
- Then you should be able to run this test